tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1530,6 +1530,36 @@ Here are some examples of expressions::
 Identifiers, attributes, and subscripts are valid assignment targets. Other expressions cannot be on the left side of
 an assignment.
 
+self
+^^^^
```

Comment:
```suggestion
~~~~
```
Maybe promote the header by one level? I don't see how `self` is semantically a subsection of `Expressions`. 

I'm also not sure if this is exactly the right section of the page to put this - is `Statements and control flow` really more fitting than either `Classes` or `Properties (setters and getters)`? 

However, since I'm not sure *exactly* where the best place to put it is, it is okay to merge this PR with the new section where it is, and consider moving it in a followup PR.

Reply:
If we ignore the specifics of how the engine treats `self` (most notably, for the engine it's a keyword and not a "name"/identifier), `self` is just another variable that refers to the current class instance, so, personally, it makes sense to me to display `self` as a special-case of an identifier expression.

If we assume that `self` does not indeed belong to `Expressions` and we should move it somewhere else, i also can't find a fitting place for it. Moving it one level up is definitely not a solution (`self` is not a statement (||unless you just write consisting of a standalone `self`, but that's a technicality||)). The best i can find is the `Classes` section, however, given it's current content, i struggle to see `self` there without rewriting parts or all of the section.

Reply:
Okay, that reasoning makes sense. If there's no other obvious place to put it, it's good here.